File: internal/models/agent.go
package models

import "time"

type AgentStatus string

const (
	AgentStatusPending AgentStatus = "pending"
	AgentStatusOnline  AgentStatus = "online"
	AgentStatusOffline AgentStatus = "offline"
	AgentStatusRevoked AgentStatus = "revoked"
)

type Agent struct {
	Base
	OrganizationID string      `json:"organizationId" gorm:"type:uuid;not null;index"`
	Name           string      `json:"name" gorm:"not null"`
	TokenHash      string      `json:"-" gorm:"uniqueIndex;not null"`
	Status         AgentStatus `json:"status" gorm:"type:varchar(20);not null;default:'pending';index"`
	Version        string      `json:"version,omitempty"`
	Hostname       string      `json:"hostname,omitempty"`
	Platform       string      `json:"platform,omitempty"`
	LastSeenAt     *time.Time  `json:"lastSeenAt,omitempty"`
	ConnectedAt    *time.Time  `json:"connectedAt,omitempty"`
	RevokedAt      *time.Time  `json:"revokedAt,omitempty"`
	Metadata       string      `json:"metadata" gorm:"type:jsonb;not null;default:'{}'"`
}

File: apps/web/src/interfaces/agent.ts
import type { Entity } from '#/interfaces/api';

export type AgentStatus = 'pending' | 'online' | 'offline' | 'revoked';

export type Agent = Entity & {
  organizationId: string;
  name: string;
  status: AgentStatus;
  version?: string;
  hostname?: string;
  platform?: string;
  lastSeenAt?: string;
  connectedAt?: string;
  revokedAt?: string;
  metadata: string;
};
